app/src/main/java/SmartTrainTools/Journey.java

```diff
@@ -449,7 +449,7 @@ public void fetchAvailability() throws IOException,AvailabilityFailure{
             return;
         /*String cd=SmartTools.generateAvailabilityHash(this.train.getNo(),this.date,this.src,this.dest,"GN",this.Tclass);
         if(Config.cachedAvailabilityStatus.containsKey(cd)){
-            this.status=Config.cachedAvailabilityStatus.get(cd);
+            this.status=Config.cachedAvailabilityStatus.getCursor(cd);
```

Comment:
abe isme refractor ki wajah se change ho gya....is file k changes undo krna padega